### PR TITLE
test: fix unstable test metrics

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 

part of https://github.com/pingcap/dm/issues/1997

### What is changed and how it works?

Reduced test instability by skipping `HeartbeatEvent` 

### Check List 

Tests 

 - Integration test

Code changes

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by Ehco1996** [comment]
@okJiang  PTAL

File: tests/metrics/run.sh [hunk 1]
```diff
@@ -95,7 +95,15 @@ function run() {
 	# restart dm-worker1
 	pkill -hup -f dm-worker1.toml 2>/dev/null || true
 	wait_pattern_exit dm-worker1.toml
-	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/changeTickerInterval=return(5);github.com/pingcap/dm/syncer/noJobInQueueLog=return()"
+
+	inject_points=(
+		"github.com/pingcap/dm/syncer/changeTickerInterval=return(5)"
+		"github.com/pingcap/dm/syncer/noJobInQueueLog=return()"
+		"github.com/pingcap/dm/syncer/IgnoreSomeTypeEvent=return(\"HeartbeatEvent)"
```
Comment (okJiang) [inline tests/metrics/run.sh:102]: Lack of a quote?

File: syncer/syncer.go [hunk 2]
```diff
@@ -1925,6 +1925,14 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 		var originSQL string // show origin sql when error, only ddl now
 		var err2 error
+
+		failpoint.Inject("IgnoreSomeTypeEvent", func(val failpoint.Value) {
+			if e.Header.EventType.String() == val.(string) {
+				tctx.L().Debug("IgnoreSomeTypeEvent", zap.Reflect("event", e))
+				failpoint.Continue()
```
Comment (okJiang) [inline syncer/syncer.go:1932]: Why choose to continue here?

Comment (Ehco1996) [reply syncer/syncer.go:1932]: just want to ignore this event and jump to next round of for-loop

**Review by lance6716 — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**